src/include/machnet_engine.h

```diff
@@ -241,6 +241,19 @@ class MachnetEngineSharedState {
     return true;
   }
 
+  int GetQueueId(const net::Ipv4::Address &ipv4_addr,
```

Comment:
A few comments for this function:

1. This function is **not** thread-safe and it can introduce a bug if a listener is being added to the `unordered_map` while another engine tries to retrieve the queue id. Please add a proper lock.
2. Returning `-1` is not preferrable in this codebase. Also, in the rest of the codebase we use `uint16_t` as the type for queue ID. Let's have this method return `std::optional<uint16_t>`. You can check `SrcPortAlloc()` method to see it's usage.